src/source/Sctp/Sctp.c

```diff
@@ -213,11 +212,52 @@ STATUS sctpSessionWriteDcep(PSctpSession pSctpSession, UINT32 streamId, PCHAR pC
      CHK(pSctpSession != NULL && pChannelName != NULL, STATUS_NULL_ARG);
      CHK((pPacket = (PBYTE) MEMCALLOC(1, pPacketSize)) != NULL, STATUS_NOT_ENOUGH_MEMORY);
 
-     pPacket[0] = DCEP_DATA_CHANNEL_OPEN;
+     /* Setting the fields of DATA_CHANNEL_OPEN message */
+
+     pPacket[0] = DCEP_DATA_CHANNEL_OPEN; // message type
+
+     // Set Channel type based on supplied parameters
      pPacket[1] = DCEP_DATA_CHANNEL_RELIABLE;
+
+     // Set channel type and reliability parameters based on input
+     // SCTP allows fine tuning the channel robustness:
+     // 1. Ordering: The data packets can be sent out in an ordered/unordered fashion
+     // 2. Reliability: This determines how the retransmission of packets is handled.
+     // There are 2 parameters that can be fine tuned to achieve this:
+     //     a. Number of retransmits
+     //     b. Packet lifetime
+     // Default values for the parameters is 0. This falls back to reliable channel
+     if(pRtcDataChannelInit != NULL) {
+         switch(pRtcDataChannelInit->ordered) {
+         case TRUE:
+             if(pRtcDataChannelInit->maxRetransmits > 0) {
```

Comment:
Optimize the boolean expressions here - when changing from switch to if/else, make sure you don't repeat the same statements in both clauses.

Reply:
Not sure if I can avoid not repeating the statements because pPacket[1] varies in all 4 cases. The only place I can think of optimizing is in `putUnalignedInt16BigEndian()`, where I can add the line after the if-else clause based on pPacket[1].